File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int flag[10005]={0}, i;
        for(i=0; i<nums.size(); i++) {
            flag[nums[i]] = 1;
        }
        for(i=0; i<nums.size()+1; i++) {
            if(flag[i] == 0){
                break;
            }
        }
        return i;
    }
};

File: 0268-missing-number/0268-missing-number.go
func missingNumber(nums []int) int {
    for i := range nums {
        for nums[i] > 0 && nums[i] <= i && nums[nums[i]-1] != nums[i] {
            nums[i], nums[nums[i]-1] = nums[nums[i]-1], nums[i]
        }
    }
    for i, v := range nums {
        if v != i+1 {
            return i+1
        }
    }
    return 0
}